android/sdl_android/src/main/java/com/smartdevicelink/managers/video/resolution/VideoStreamingRange.java

```diff
@@ -0,0 +1,98 @@
+package com.smartdevicelink.managers.video.resolution;
+
+import com.smartdevicelink.proxy.rpc.ImageResolution;
+
+public class VideoStreamingRange {
+    private Resolution minimumResolution;
+    private Resolution maximumResolution;
+    private Double minimumDiagonal;
+    private Double minimumAspectRatio;
+    private Double maximumAspectRatio;
+
+    public VideoStreamingRange(Resolution minimumResolution, Resolution maximumResolution, Double minimumDiagonal, Double minimumAspectRatio, Double maximumAspectRatio) {
+        this.minimumResolution = minimumResolution;
+        this.maximumResolution = maximumResolution;
+        this.minimumDiagonal = minimumDiagonal;
+        this.minimumAspectRatio = minimumAspectRatio;
+        this.maximumAspectRatio = maximumAspectRatio;
+    }
+
+    private VideoStreamingRange() {
+    }
+
+    public Resolution getMinimumResolution() {
+        return minimumResolution;
+    }
+
+    public Resolution getMaximumResolution() {
+        return maximumResolution;
+    }
+
+    public Double getMinimumDiagonal() {
+        return minimumDiagonal;
+    }
+
+    public Double getMinimumAspectRatio() {
+        return minimumAspectRatio;
+    }
+
+    public Double getMaximumAspectRatio() {
+        return maximumAspectRatio;
+    }
+
+    public VideoStreamingRange setMinSupportedResolution(Resolution minSupportedResolution) {
+        this.minimumResolution = minSupportedResolution;
+        return this;
+    }
+
+    public VideoStreamingRange setMaxSupportedResolution(Resolution maxSupportedResolution) {
+        this.maximumResolution = maxSupportedResolution;
+        return this;
+    }
+
+    public VideoStreamingRange setMinimumDiagonal(Double minimumDiagonal) {
+        this.minimumDiagonal = minimumDiagonal;
+        return this;
+    }
+
+    public VideoStreamingRange setMinimumAspectRatio(Double minimumAspectRatio) {
+        this.minimumAspectRatio = minimumAspectRatio;
+        return this;
+    }
+
+    public VideoStreamingRange setMaximumAspectRatio(Double maximumAspectRatio) {
+        this.maximumAspectRatio = maximumAspectRatio;
+        return this;
+    }
+
+    public Boolean isImageResolutionInRange(ImageResolution currentResolution) {
+
```

Comment:
A few additional checks should be made here,

if the passed resolution is null we should return false (no resolution received from the capability)
if both the minimumResolution and maximumResolution are null we should return true (the developer does not want to use resolution for the range)
If one of the reolutions (minimumResolution and maximumResolution) is null or set to 0 values it should return false (only one of the resolutions is null or the resolution has 0 values)

```
        if (currentResolution == null) {
            return false;
        }

        if (minimumResolution == null && maximumResolution == null) {
            return true;
        }

        if (minimumResolution == null || (minimumResolution.getResolutionWidth() == 0 && minimumResolution.getResolutionHeight() == 0)) {
            return false;
        }

        if (maximumResolution != null || (maximumResolution.getResolutionWidth() == 0 && maximumResolution.getResolutionHeight() == 0)) {
            return false;
        }
```